onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc

```diff
@@ -87,19 +87,20 @@ static Status RegisterMIGraphXKernels(KernelRegistry& kernel_registry) {
 }
 
 static std::shared_ptr<KernelRegistry> s_kernel_registry;
-void Shutdown_DeleteRegistry() {
+
+void InitializeRegistry() {
+  s_kernel_registry = KernelRegistry::Create();
+  auto status = RegisterMIGraphXKernels(*s_kernel_registry);
```

Comment:
this one is different from the other EPs, for example:
[https://github.com/microsoft/onnxruntime/blob/6c4887661a1b5d5088277b19085bea7e10acd251/onnxruntime/core/providers/cuda/cuda_execution_provider.cc#L2208-L2211](https://github.com/microsoft/onnxruntime/blob/6c4887661a1b5d5088277b19085bea7e10acd251/onnxruntime/core/providers/cuda/cuda_execution_provider.cc#L2208-L2211)
can they all be consistent? is it important that s_kernel_registry is reset if there is an error?

Reply:
I'll see if I can make the error handling just fail to load the provider. That seems the cleanest. I'll also make this one more consistent.

Reply:
Done